File: microstartercli/src/main/resources/micronaut/functions/azure/java/SaveRequestHander.java
package com.example;
import com.microsoft.azure.functions.HttpStatus;
import com.microsoft.azure.functions.ExecutionContext;
import com.microsoft.azure.functions.HttpRequestMessage;
import com.microsoft.azure.functions.HttpMethod;
import com.microsoft.azure.functions.HttpResponseMessage;
import com.microsoft.azure.functions.annotation.AuthorizationLevel;
import com.microsoft.azure.functions.annotation.FunctionName;
import com.microsoft.azure.functions.annotation.HttpTrigger;
import io.micronaut.azure.function.AzureFunction;
import java.util.Optional;


public class Function extends AzureFunction {
    @FunctionName("HttpTrigger")
    public HttpResponseMessage hello(
            @HttpTrigger(
                    name = "req",
                    methods = {HttpMethod.GET},
                    route = "{*route}",
                    authLevel = AuthorizationLevel.ANONYMOUS)
                    HttpRequestMessage<Optional<String>> request,
            ExecutionContext context) {
        if (context != null) {
            context.getLogger().info("Executing Function: " + getClass().getName());
        }
        return request.createResponseBuilder(HttpStatus.OK).body("Hello World").build();
    }

}

File: microstartercli/src/main/resources/micronaut/functions/azure/kotlin/SaveRequestHander.kt
package com.example;

import com.microsoft.azure.functions.HttpStatus
import com.microsoft.azure.functions.ExecutionContext
import com.microsoft.azure.functions.HttpRequestMessage
import com.microsoft.azure.functions.HttpMethod
import com.microsoft.azure.functions.HttpResponseMessage
import com.microsoft.azure.functions.annotation.AuthorizationLevel
import com.microsoft.azure.functions.annotation.FunctionName
import com.microsoft.azure.functions.annotation.HttpTrigger
import io.micronaut.azure.function.AzureFunction
import java.util.Optional

class Function : AzureFunction() {
    @FunctionName("HttpTrigger")
    fun hello(
        @HttpTrigger(
            name = "req",
            methods = [HttpMethod.GET],
            route = "{*route}",
            authLevel = AuthorizationLevel.ANONYMOUS
        ) request: HttpRequestMessage<Optional<String?>?>,
        context: ExecutionContext?
    ): HttpResponseMessage {
        context?.logger?.info("Executing Function")
        return request.createResponseBuilder(HttpStatus.OK).body("Hello World").build()
    }
}